src/main/java/org/spongepowered/common/mixin/core/server/management/MixinPlayerInteractionManager.java

```diff
@@ -523,4 +527,29 @@ public EnumActionResult processRightClick(EntityPlayer player, net.minecraft.wor
         }
     }
 
+    private final Deque<CauseStackManager.StackFrame> tryHarvestBlockStackFrames = new LinkedList<>();
+
+    /**
+     * Some mods (e.g. OreExcavator) will call tryHarvestBlock directly, and
+     * the cause stack may not have the player on it. This should fix that.
+     * See SpongeForge#2789.
+     */
+    @Inject(method = "tryHarvestBlock", at = @At("HEAD"))
+    private void onTryHarvestBlockHead(BlockPos pos, CallbackInfoReturnable<Boolean> cir) {
+        if (!Sponge.getCauseStackManager().getCurrentCause().contains(this.player)) {
```

Comment:
Considering this code came about due to tick logic, I would rather just throw it on the stack and pop it. Worst case, we have a dupe object. Best case we don't. Either case, we save a lookup.

Reply:
Done.